### PR TITLE
implemented new header design

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
__Tasks:__
- [x] implemented new header design
- [x] moved tabs behaviour into header
- [ ] got feedback

__PRs:__
- Husky: https://github.com/massiveart/husky/pull/512
- Sulu-Buttons and behat tests: https://github.com/sulu-io/sulu/pull/1421

__Informations:__

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tests pass?   | none
| Fixed tickets | none
| Doc           | none

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * FEATURE     #1421 [AdminBundle]     new button-api for header and tabs
```
Comment (wachterjohannes) [inline CHANGELOG.md]: first letter uppercase

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,23 @@
 
 ## Dev-develop
 
+### Header-toolbar and tabs
+The header got a complete redesign, the breadcrumb, title and bottom-content are not available anymore. Also the event header.set-toolbar got removed. A sulu-header can still be initialized via the header property when a view-component gets started.
```
Comment (wachterjohannes) [inline UPGRADE.md]: place event names in ``

File: UPGRADE.md [hunk 3]
```diff
@@ -2,6 +2,23 @@
...
+The language-changer can be configured as it was. Template and parentTemplate in contrast are not supported anymore. Instead you pass an array of sulu-buttons.
+Moreover the format of the buttons itself changed: https://github.com/massiveart/husky/blob/f9b3abeb547553c9c031710f1f98d0288b08ca9c/UPGRADE.md
+
+#### Sulu-buttons
+Buttons for toolbars get specified in an aura-extension (sandbox.sulu.buttons and sandbox.sulu.buttons.dropdownItems). Therfore each bundle can add their own buttons to the pool. The toolbar in the header fetches its buttons from this pool.
+
+#### List-toolbar
+The inHeader option got removed and is not supported anymore. Sulu.buttons are used internally and can be passed via the template which is recommended instead of using string templates.
+
+
```
Comment (wachterjohannes) [inline UPGRADE.md]: remove this empty line

File: UPGRADE.md [hunk 4]
```diff
@@ -2,6 +2,23 @@
...
+#### Toolbar
+The language-changer can be configured as it was. Template and parentTemplate in contrast are not supported anymore. Instead you pass an array of sulu-buttons.
+Moreover the format of the buttons itself changed: https://github.com/massiveart/husky/blob/f9b3abeb547553c9c031710f1f98d0288b08ca9c/UPGRADE.md
+
+#### Sulu-buttons
+Buttons for toolbars get specified in an aura-extension (sandbox.sulu.buttons and sandbox.sulu.buttons.dropdownItems). Therfore each bundle can add their own buttons to the pool. The toolbar in the header fetches its buttons from this pool.
+
+#### List-toolbar
+The inHeader option got removed and is not supported anymore. Sulu.buttons are used internally and can be passed via the template which is recommended instead of using string templates.
```
Comment (wachterjohannes) [inline UPGRADE.md]: `Sulu.buttons`

File: UPGRADE.md [hunk 5]
```diff
@@ -2,6 +2,23 @@
...
+
+#### Tabs
+The tabs can be configured with the 'url', 'data' and 'container' option. The 'option' fullControll' got removed. You can get the same effect by passing data with no 'component'-property.
+
+#### Toolbar
+The language-changer can be configured as it was. Template and parentTemplate in contrast are not supported anymore. Instead you pass an array of sulu-buttons.
+Moreover the format of the buttons itself changed: https://github.com/massiveart/husky/blob/f9b3abeb547553c9c031710f1f98d0288b08ca9c/UPGRADE.md
+
+#### Sulu-buttons
+Buttons for toolbars get specified in an aura-extension (sandbox.sulu.buttons and sandbox.sulu.buttons.dropdownItems). Therfore each bundle can add their own buttons to the pool. The toolbar in the header fetches its buttons from this pool.
```
Comment (wachterjohannes) [inline UPGRADE.md]: `sandbox.sulu.buttons` and `sandbox.sulu.buttons.dropdownItems`

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ define(function() {
          * raised after the initialization has finished
          * @event sulu.app.initialized
          */
-        INITIALIZED = function() {
+        INITIALIZED = function () {
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js]: no space after function

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js [hunk 7]
```diff
@@ -259,37 +265,28 @@ define(function() {
...
+            router.navigate(route, {trigger: trigger});
             this.sandbox.dom.scrollTop(this.sandbox.dom.$window, 0);
         },
 
         /**
          * Cleans things up before navigating
-         * @param {String} route
          */
-        beforeNavigateCleanup: function(route) {
-            this.currentRoute = route;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js]: are you sure nobody uses this?

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/header/main.js [hunk 8]
```diff
@@ -746,6 +526,43 @@ define([], function () {
         },
 
         /**
+         * Renderes and starts the language-changer dropdown
+         */
+        startLanguageChanger: function() {
+            if (!!this.options.toolbarLanguageChanger) {
+                var $element = this.sandbox.dom.createElement(this.sandbox.util.template(templates.languageChanger)({
+                    title: this.options.toolbarLanguageChanger.preSelected || this.sandbox.sulu.user.locale
+                })),
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/header/main.js]: intention is wrong

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/header/main.js [hunk 9]
```diff
@@ -882,147 +739,107 @@ define([], function () {
...
+         * //todo: make it cleaner!
+         * @param type {string} "edit" or "add"
+         * @param saved {boolean} false if the toolbar should represent a dirty-state
+         * @param highlight {boolean} true to change with a highlight effect
          */
-        hide: function () {
-            this.sandbox.dom.addClass(this.$el, 'invisible');
-            this.sandbox.dom.hide(constants.headerBackgroundSelector);
-        },
+        changeToolbarState: function (type, saved, highlight) {
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/header/main.js]: remove this empty line

File: src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/base.scss [hunk 10]
```diff
@@ -38,3 +38,10 @@ input, select {
 .m-top-1 {
     margin-top: 1px !important;
 }
+
+// List-container
+/*----------------------------------------*/
+
+.list-toolbar {
+    padding-bottom: 20px;
+}
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/base.scss]: add an empty line at the end of the file

File: src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/layout.scss [hunk 11]
```diff
@@ -201,7 +237,9 @@
...
         float: none;
-        width: auto;
-        max-width: $contentFixedWidth;
+        #content {
+            max-width: $contentFixedWidth;
+            width: auto;
+        }
     }
-}
+}
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/layout.scss]: empty line at the end of the file

File: src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/main.js [hunk 12]
```diff
@@ -252,11 +252,11 @@ define(['sulucategory/model/category',
 
         /**
          * Renders the from for add and edit
-         * @param locale {String} the locale in which the category gets loaded. can be undefined
+         * @param language {Object} the language object from the header language changer
          */
-        renderForm: function (locale) {
+        renderForm: function (language) {
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/main.js]: no space after function

File: src/Sulu/Bundle/LocationBundle/Resources/public/js/main.js [hunk 13]
```diff
@@ -33,7 +33,7 @@ define({
         // sandbox.mvc.routes.push({
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/LocationBundle/Resources/public/js/main.js]: remove this .. i think this shouldn't be here (:

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/main.js [hunk 14]
```diff
@@ -46,19 +44,7 @@ define({
...
             callback: function(id, content) {
-                this.html(
-                    '<div data-aura-component="collections/components/content@sulumedia" data-aura-content="' + content + '" data-aura-id="' + id + '"/>'
-                );
-            }
-        });
-
-        // show a single collection with files and upload
-        sandbox.mvc.routes.push({
-            route: 'media/collections/edit::id/:content/edit::mediaId',
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/main.js]: this route is used by the search for deeplinking the media-result

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js (hunk 7 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js]: it's still there, just moved it

File: CHANGELOG.md (hunk 1 above)
Comment (marcelmoosbrugger) [inline CHANGELOG.md]: fixed

File: UPGRADE.md (hunk 2 above)
Comment (marcelmoosbrugger) [inline UPGRADE.md]: fixed

File: UPGRADE.md (hunk 3 above)
Comment (marcelmoosbrugger) [inline UPGRADE.md]: fixed

File: UPGRADE.md (hunk 4 above)
Comment (marcelmoosbrugger) [inline UPGRADE.md]: fixed

File: UPGRADE.md (hunk 5 above)
Comment (marcelmoosbrugger) [inline UPGRADE.md]: fixed

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js (hunk 6 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/app/main.js]: fixed

File: src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/base.scss (hunk 10 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/AdminBundle/Resources/public/scss/modules/base.scss]: fixed

**Pull request merged**